packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts

```diff
@@ -100,6 +108,7 @@ export class ApplicationLoadBalancer extends BaseLoadBalancer implements IApplic
 
     if (props.http2Enabled === false) { this.setAttribute('routing.http2.enabled', 'false'); }
     if (props.idleTimeout !== undefined) { this.setAttribute('idle_timeout.timeout_seconds', props.idleTimeout.toSeconds().toString()); }
+    if (props.dropInvalidHeaderFields) {this.setAttribute('routing.http.drop_invalid_header_fields.enabled', 'true'); }
```

Comment:
Actually, before merging this I wonder if we could set `true` as the default. Do you know of any downsides to setting this to `true` by default?

Reply:
I don't think there are any downsides if you are deploying a new application and still flexible about your custom header fields. However, since existing applications will also get affected, I think it is safe to leave the default to `false`.

For example, header names with underscores are dropped if this attribute is set to `true`.
https://stackoverflow.com/questions/58848623/what-does-alb-consider-a-valid-header-field